src/DynamoCore/Models/NodeModel.cs

```diff
@@ -1612,19 +1611,19 @@ internal void RequestValueUpdateAsync(IScheduler scheduler, EngineController eng
                 VariableName = variableName
             });
 
-            task.Completed += OnNodeValueQueried;
+            task.Completed += QueryMirrorDataAsyncTaskCompleted;
             scheduler.ScheduleForExecution(task);
         }
 
-        private void OnNodeValueQueried(AsyncTask asyncTask)
+        private void QueryMirrorDataAsyncTaskCompleted(AsyncTask asyncTask)
         {
-            lock (cachedMirrorDataMutex)
+            asyncTask.Completed -= QueryMirrorDataAsyncTaskCompleted;
+
+            var task = asyncTask as QueryMirrorDataAsyncTask;
+            if (task != null)
             {
-                var task = asyncTask as QueryMirrorDataAsyncTask;
-                cachedMirrorData = task.MirrorData;
+                this.CachedValue = task.MirrorData;
             }
```

Comment:
This changes a few things:

1. We now actually clean up the event registration
2. We check the task is of the correct type (unlikely but possible)
3. We remove lock and use the `CachedValue` setter that correctly locks and emits the appropriate `CachedValue` PropertyChanged event.